internal/services/portfolio/balance_service.go

```diff
@@ -375,11 +377,64 @@ func (s *BalanceService) fetchUserBalance(wallet string) (dtos.Balance, error) {
 		ZerolossUnrealizedPnl: zerolossUnrealizedPnl.InexactFloat64(),
 		ZerolossRealizedPnl:   zerolossRealizedPnl.InexactFloat64(),
 		ZerolossVolumeTraded:  zerolossVolumeTraded.InexactFloat64(),
+		CurrentAPY:            apy.InexactFloat64(),
 	}
 
 	return userBalance, nil
 }
 
+func (s *BalanceService) CalculateCurrentAPY(wallet string, totalBalance decimal.Decimal) decimal.Decimal {
+	/* fetch total balance and average expiry from database */
+	zerolossBalanceDetails, err := s.repo.GetTotalFundsAvgExpiryForUser(s.ctx, utils.PgText(wallet))
+	if err != nil {
+		return decimal.Zero
+	}
+
+	totalFunds, err := utils.PgNumericToDecimal(zerolossBalanceDetails.TotalBalance)
+	if err != nil {
+		return decimal.Zero
+	}
+
+	// If there are no funds, return 0 APY
+	if totalFunds.IsZero() {
+		return decimal.Zero
+	}
+
+	// Get average expiry in days
+	avgExpiryDays, err := utils.PgNumericToDecimal(zerolossBalanceDetails.AvgDaysTillMaturity)
+	if err != nil {
+		return decimal.Zero
+	}
+
+	// If expiry is 0 or negative, return 0 APY
+	if avgExpiryDays.LessThanOrEqual(decimal.Zero) {
+		return decimal.Zero
+	}
+
+	/* calculate current apy
+	   Current APY :- Total Zeroloss Balance / Total Zeroloss Added funds * 365/Average Expiry
+	*/
+
+	// Calculate ratio of current balance to total added funds
+	balanceRatio := totalBalance.Div(totalFunds)
+
+	// Calculate 365/Average Expiry
+	daysRatio := decimal.NewFromInt(365).Div(avgExpiryDays)
+
+	// Calculate final APY
+	apy := balanceRatio.Mul(daysRatio)
+
+	// Convert to percentage (multiply by 100)
+	apy = apy.Mul(decimal.NewFromInt(100))
+
+	// Ensure APY is not negative
+	if apy.LessThan(decimal.Zero) {
+		return decimal.Zero
+	}
+
+	return apy
+}
```

Comment:
I think the above code is unnecessarily verbose and can be written in much better way 
```suggestion
func (s *BalanceService) CalculateCurrentAPY(wallet string, totalBalance decimal.Decimal) decimal.Decimal {
	// Fetch balance details from DB.
	details, err := s.repo.GetTotalFundsAvgExpiryForUser(s.ctx, utils.PgText(wallet))
	if err != nil {
		return decimal.Zero
	}

	totalFunds, err := utils.PgNumericToDecimal(details.TotalBalance)
	if err != nil || totalFunds.IsZero() {
		return decimal.Zero
	}

	avgExpiry, err := utils.PgNumericToDecimal(details.AvgDaysTillMaturity)
	if err != nil || avgExpiry.LessThanOrEqual(decimal.Zero) {
		return decimal.Zero
	}

	// Calculate APY:
	// APY = (totalBalance / totalFunds) * (365 / avgExpiry) * 100
	apy := totalBalance.
		Div(totalFunds).
		Mul(decimal.NewFromInt(365).Div(avgExpiry)).
		Mul(decimal.NewFromInt(100))

	if apy.LessThan(decimal.Zero) {
		return decimal.Zero
	}

	return apy
}
```